src/util.ts

```diff
@@ -0,0 +1,25 @@
+export const compareFloats = (
```

Comment:
Because of the 0.01 factor, I would consider naming this a little more specifically. Like compareTimes(time1, time2, timeStepSize) and the description would be `compares two times by testing to see whether they are within some small fraction of the time step size`.   The last 3 lines of the function are the true `compareFloats` but the epsilon multiplier makes it different.
Another way could be to pass the multiplier as a 4th argument but provide a default value:
compareTimes(time1, time2, timeStepSize, stepSizeFraction=0.01)
Just a suggestion. See what you think.

Reply:
Gotcha. I was trying to figure out the best place to put a global constant for that multiplier value and finally decided the best way to safeguard it from getting changed was to hard-code it inside the function 😆 I see what you're saying though. I'll turn it into a 4th argument with a default value.